src/server/src/Server.cpp

```diff
@@ -253,37 +291,41 @@ Server::State Server::get_state()
     return state;
 }
 
-
 void Server::stop_motion()
 {
     drdStop();
 }
 
-
-void Server::set_state(const State& state)
+void Server::set_state(const State &state)
 {
     mutex_.lock();
     state_ = state;
     mutex_.unlock();
 }
 
-
 void Server::stream_robot_state()
 {
-    double position[3];
+    double position[3]; // position px, py, pz
 
     dhdGetPosition(&position[0], &position[1], &position[2]);
 
-    Vector& position_out = port_position_.prepare();
+    Vector &position_out = port_position_.prepare();
     position_out = Vector(3, position);
     port_position_.write();
 
+    double velocity[3]; // velocity vx, vy, vz
+
+    dhdGetLinearVelocity(&velocity[0], &velocity[1], &velocity[2]);
+
+    Vector &velocity_out = port_velocity_.prepare();
+    velocity_out = Vector(3, velocity);
+    port_velocity_.write();
 
-    double force[3];
+    double force[3]; // force fx, fy, fz
 
     dhdGetForce(&force[0], &force[1], &force[2]);
 
-    Vector& force_out = port_force_.prepare();
+    Vector &force_out = port_force_.prepare();
```

Comment:
```suggestion
    Vector& force_out = port_force_.prepare();
```

Reply:
wow, and even with that error it was working???

Reply:
It is not an error, it is just yet another convention :)

Reply:
I have to check again. Update this asap.